Tests/UnitTests/Purchasing/Purchases/PurchasesConfiguringTests.swift

```diff
@@ -362,6 +362,92 @@ class PurchasesConfiguringTests: BasePurchasesTests {
+                                         customEntitlementsComputation: true)
+        self.setupPurchases()
+
+        expect(self.purchasesDelegate.customerInfoReceivedCount).toEventually(equal(0))
+    }
+
+    func testWithoutCustomEntitlementComputationDoesntSkipFirstDelegateCall() throws {
+        self.systemInfo = MockSystemInfo(finishTransactions: true,
+                                         customEntitlementsComputation: false)
+        self.setupPurchases()
+
+        expect(self.purchasesDelegate.customerInfoReceivedCount).toEventually(equal(1))
+    }
+
+    func testConfigureWithCustomEntitlementComputationFatalErrorIfNoAppUserID() throws {
+        self.systemInfo = MockSystemInfo(finishTransactions: true,
+                                         customEntitlementsComputation: true)
+
+        let expectedMessage = Strings.configure.custom_entitlements_computation_enabled_but_no_app_user_id.description
+
+        expectFatalError(expectedMessage: expectedMessage, testcase: {
+            self.setupAnonPurchases()
+        })
+    }
+
+    func testConfigureWithCustomEntitlementComputationNoFatalErrorIfAppUserIDPassedIn() throws {
+        self.systemInfo = MockSystemInfo(finishTransactions: true,
+                                         customEntitlementsComputation: true)
+        expectNoFatalError(testcase: {
+            self.setupPurchases()
+        })
+    }
+
+    func testConfigureWithCustomEntitlementComputationLogsInformationMessage() throws {
+        self.systemInfo = MockSystemInfo(finishTransactions: true,
+                                         customEntitlementsComputation: true)
+        let logger = TestLogHandler()
+
+        self.setupPurchases()
+
+        logger.verifyMessageWasLogged(Strings.configure.custom_entitlements_computation_enabled)
+    }
+
+    func testConfigureWithoutCustomEntitlementComputationDoesntLogInformationMessage() throws {
+        let logger = TestLogHandler()
+
+        self.setupPurchases()
+
+        logger.verifyMessageWasNotLogged(Strings.configure.custom_entitlements_computation_enabled)
+    }
+
+    func testConfigureWithCustomEntitlementComputationDisablesLogOut() throws {
+        self.systemInfo = MockSystemInfo(finishTransactions: true,
+                                         customEntitlementsComputation: true)
+        self.setupPurchases()
+
+        var receivedCustomerInfo: CustomerInfo?
+        var receivedError: PublicError?
+        var completionCalled = false
+
+        self.purchases.logOut { customerInfo, error in
+            completionCalled = true
+            receivedCustomerInfo = customerInfo
+            receivedError = error
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        expect(receivedCustomerInfo).to(beNil())
+        let unwrappedError = try XCTUnwrap(receivedError)
+        expect(unwrappedError.code).to(equal(
+            ErrorUtils.featureNotAvailableInCustomEntitlementsComputationModeError().errorCode))
```

Comment:
Maybe compare the error and not just the code:
```suggestion
        expect(unwrappedError).to(matchError(
            ErrorUtils.featureNotAvailableInCustomEntitlementsComputationModeError()
         ))
```

Reply:
I had started by doing that, but then I got differences in the error since they point to different lines of code, and figured this was a good enough proxy

Reply:
Oh, how about 
```swift
  expect(unwrappedError).to(matchError(
            ErrorUtils.featureNotAvailableInCustomEntitlementsComputationModeError().asPublicError
         ))
```

The main reason I'm suggesting this is because you're only comparing the `code` and not the `domain`.

Reply:
same problem, that's actually the exact code I'd started with. 

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/3922667/234036889-d1b240ad-099d-4fee-92c3-e63942e3df63.png">


I can add another check for the domain, though

Reply:
added. I'm merging for now, happy to update in a follow-up if there's another easy way to do this

Reply:
See my reply :)

Reply:
ohh nice I'd missed that, needed a page refresh 😅  I'll try that